### PR TITLE
Correct comparison of defaults for String type columns in PostgreSQL

### LINKED ISSUE #1587: SchemaUtils.addMissingColumnsStatements() fails to compare varchar and text column with empty default value. Postgres dialect used
If in schema definition there are fields with empty default values like these
`val varchar = varchar(“varchar”, 255).default(“”)`
`val text = text(“text”).default(“”)`
then SchemaUtils.createMissingTablesAndColumns() triggers column alteration for columns of type varchar or text

The problem is that the value obtained from the schema is not equal to the default value:
https://github.com/JetBrains/Exposed/blob/c04cac255fd4c946f763f29df8062394b1787c4e/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt#L210-L211

For varchar type `existingCol.defaultDbValue` returns "::character varying" and `col.dbDefaultValue?.let { dataTypeProvider.dbDefaultToString(it) }` returns "'::character varying"
There is an extra single quote

For text type `existingCol.defaultDbValue returns` "::text" and `col.dbDefaultValue?.let { dataTypeProvider.dbDefaultToString(it) }` returns "'::character varying"
Here the type is also not defined correctly, for all string types the corresponding varchar
https://github.com/JetBrains/Exposed/blob/b9b53f8bbdfbf8cbab56d5602f92543e2ccd473c/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt#L154-L157

Reproduce on 0.38.2 and 0.39.2

### PR DESCRIPTION
**Pull request by AlexeySoshin**
Fixes #1587

Stripping the first single quote in PostgreSQL doesn't seem like a good strategy, since `'B'::text` is a valid default value.

Branching with Text/Varchar seems necessary, because the default expression is always of `TextColumnType`, as it is mapped from `String` literal:

https://github.com/JetBrains/Exposed/blob/9b561b02971b13e7445aa37223dbfabfd419d14b/exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Op.kt#L567

### CODE REVIEW

File: exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt [hunk 1]
```diff
@@ -172,6 +172,7 @@ class JdbcDatabaseMetadataImpl(database: String, val metadata: DatabaseMetaData)
         is SQLServerDialect -> defaultValue.trim('(', ')', '\'')
         is OracleDialect -> defaultValue.trim().trim('\'')
         is MysqlDialect -> defaultValue.substringAfter("b'").trim('\'').trim()
+        is PostgreSQLDialect -> defaultValue.trim()
```
Comment (Tapac) [inline exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt:175]: Looks like trim here and in default branch are wrong. With it it's impossible to set default value with whitespaces (like "  ").
Could you please remove the trim and adjust your test by replacing empty string with whitespaces?

**Review by Tapac — CHANGES_REQUESTED** [review]

File: exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt (hunk 1 above)
Comment (AlexeySoshin) [reply exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt:175]: Thanks for pointing this out, I'll add another test to reproduce and will fix this.

File: exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/CreateMissingTablesAndColumnsTests.kt [hunk 2]
```diff
@@ -255,6 +255,51 @@ class CreateMissingTablesAndColumnsTests : DatabaseTestsBase() {
...
+            val varchar = varchar("varchar_column", 255).default(" ")
+            val text = text("text_column").default(" ")
+        }
+
+        val tableEmptyStringDefault = object : Table("varchar_test") {
+            val varchar = varchar("varchar_column", 255).default("")
+            val text = text("text_column").default("")
+        }
+
+        // MySQL doesn't support default values on text columns, hence excluded
```
Comment (AlexeySoshin) [inline exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/CreateMissingTablesAndColumnsTests.kt:289]: @Tapac Would appreciate your input on that

File: detekt.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # https://github.com/detekt/detekt/blob/main/detekt-core/src/main/resources/default-detekt-config.yml
 
 build:
-    maxIssues: 52
+    maxIssues: 200
```
Comment (AlexeySoshin) [inline detekt.yml:4]: @Tapac raised this as current error count is 143, and I think Detekt errors should be addressed in a different PR

Comment (Tapac) [reply detekt.yml:4]: May I ask you to merge master into the branch? Detekt issue was solved there.

Comment (AlexeySoshin) [reply detekt.yml:4]: @Tapac thank you, I'll do that

**Pull request merged**